pkg/driftctl.go

```diff
@@ -149,19 +149,33 @@ func (d DriftCTL) Run() (*analyser.Analysis, error) {
 		}
 	}
 
-	analysis, err := d.analyzer.Analyze(remoteResources, resourcesFromState)
+	analysis := d.analyzer.CompareEnumeration(remoteResources, resourcesFromState)
 	if err != nil {
 		return nil, err
 	}
 
+	if d.opts.Deep {
+		remoteResources, err = d.readResources(analysis)
+		if err != nil {
+			return nil, err
+		}
+
+		d.analyzer.CompareDetails(remoteResources, resourcesFromState)
+	}
+	d.analyzer.SetAlerts()
```

Comment:
this looks really strange to me as you use an old return analysis that will be modified by this call...